### PR TITLE
Actions fieldname corrected

### PR DESCRIPTION
**Pull request by shimisho**
The following modifications have been made for SSS Approval Request API SDK.
- The response attribute "action" has been renamed to "actions".
- Corrected the data type of the field "Body" in the "actions" structure.

### CODE REVIEW

File: ecl/sss/v1/approval_requests/results.go [hunk 1]
```diff
@@ -6,11 +6,11 @@ import (
...
-	Service string `json:"service"`
-	Region  string `json:"region"`
-	APIPath string `json:"api_path"`
-	Method  string `json:"method"`
-	Body    string `json:"body"`
+	Service string                 `json:"service"`
+	Region  string                 `json:"region"`
+	APIPath string                 `json:"api_path"`
+	Method  string                 `json:"method"`
+	Body    map[string]interface{} `json:"body"`
```
Comment (shimisho) [inline ecl/sss/v1/approval_requests/results.go]: According to the API reference, the data type of the Body is "String", but it actually contains "Object". (Key = resource name, value = json object of Request Body when creating resource.)
API Referrence: https://ecl.ntt.com/en/documents/api-references/sss/rsts/APIList/Approval/GetApprovalRequest.html
↓Get Approval Request API Response Body
```
{"request_id":"req0000075648","external_request_id":"da3cfc00-5154-11ea-8088-525400060300","approver_type":"tenant_owner","approver_id":"d2f19c353e6d4c519e530c6a78438b33","request_user_id":"ecid0000005132","service":"provider-connectivity","actions":[{"service":"provider-connectivity","region":"lab3ec","api_path":"/v2.0/tenant_connections_requests","method":"POST","body":{"tenant_connection_request":{"description":"test_desc1","name":"test_name1","network_id":"628aee34-fff9-41ad-82e2-cc25b947dad7","tags":{"test_tags1":"test1"},"tenant_id_other":"d2f19c353e6d4c519e530c6a78438b33"}}}],"descriptions":[{"lang":"en","text":"User (5e86848fbc63403daaeffc1b76b3a784) is requesting for source tenant (fa68e8d56ff7468d8fb1b056bf13bf48) to share logical-network (628aee34-fff9-41ad-82e2-cc25b947dad7) of destination tenant (d2f19c353e6d4c519e530c6a78438b33). Please check terms (https://ecl.ntt.com/files/ecl_riyoukiyaku.pdf) before accepting this request."}],"request_user":true,"approver":true,"approval_deadline":"2020-03-18 07:12:29","approval_expire":"2020-03-18 07:12:35","registered_time":"2020-02-17 07:12:30","updated_time":"2020-02-17 07:12:35","status":"approved"}
```

File: ecl/sss/v1/approval_requests/testing/fixtures.go [hunk 2]
```diff
@@ -21,13 +21,22 @@ var listResponse = fmt.Sprintf(`
       "approver_id":"11a98bf9cb144af5a204c9da566d2bd0",
       "request_user_id":"ecid9999888881",
       "service":"network",
-      "action" : [
+      "actions" : [
```
Comment (kazyshr) [inline ecl/sss/v1/approval_requests/testing/fixtures.go]: Tab and space are mixed in this file. Can you use only tab?

File: ecl/sss/v1/approval_requests/testing/fixtures.go [hunk 3]
```diff
@@ -21,13 +21,22 @@ var listResponse = fmt.Sprintf(`
       "approver_id":"11a98bf9cb144af5a204c9da566d2bd0",
```
Comment (kazyshr) [inline ecl/sss/v1/approval_requests/testing/fixtures.go]: const `startTime` is never used. Please check it.

File: ecl/sss/v1/approval_requests/results.go (hunk 1 above)
Comment (kazyshr) [inline ecl/sss/v1/approval_requests/results.go]: Can you consider using `json.RawMessage`? It is more suitable for this field than map.
https://golang.org/pkg/encoding/json/#RawMessage

**Review by kazyshr — CHANGES_REQUESTED** [review]
Commented. I suggest using `json.RawMessage`.

File: ecl/sss/v1/approval_requests/testing/fixtures.go (hunk 2 above)
Comment (shimisho) [reply ecl/sss/v1/approval_requests/testing/fixtures.go]: I didn't notice. Fixed.

File: ecl/sss/v1/approval_requests/results.go (hunk 1 above)
Comment (shimisho) [reply ecl/sss/v1/approval_requests/results.go]: I agree.I was worried that actions.body could contain strings as well as json objects.
Json.RawMessage worked fine even for strings.

**Comment by shimisho** [comment]
Comment confirmed.Check out the changes I made.

**Review by ghost — APPROVED** [review]
LGTM:)

File: ecl/sss/v1/approval_requests/testing/fixtures.go [hunk 4]
```diff
@@ -9,72 +9,101 @@ import (
 const idApprovalRequest1 = "9a76dca6-d8cd-4391-aac6f-2ea052f10f4"
 const idApprovalRequest2 = "fc578e8b-dea2-4f8c-aa7e-9026fa173632"
 
-const startTime = "2018-07-26 08:40:01"
+var actionsBody = []byte(
```
Comment (kazyshr) [inline ecl/sss/v1/approval_requests/testing/fixtures.go]: It looks mess to me.
```suggestion
var actionsBody = json.RawMessage(fmt.Sprint(
	`{`,
	`"firewall":`,
	`{`,
	`"availability_zone":"zone1-groupa",`,
	`"default_gateway":"",`,
	`"description":"abcdefghijklmnopqrstuvwxyz",`,
	`"firewall_plan_id":"bd12784a-c66e-4f13-9f72-5143d64762b6",`,
	`"name":"abcdefghijklmnopqrstuvwxyz",`,
	`"tenant_id":"6a156ddf2ecd497ca786ff2da6df5aa8"`,
	`}`,
	`}`,
))
```

File: ecl/sss/v1/approval_requests/testing/fixtures.go [hunk 5]
```diff
@@ -9,72 +9,101 @@ import (
...
-      "request_user": false,
-      "approver": true,
-      "approval_deadline": "2016-12-25 09:45:22",
-      "approval_expire": null,
-      "registered_time": "2016-12-13 02:20:21",
-      "updated_time": null,
-      "status": "expired"
-    }
-  ]
+	"approval_requests": [
```
Comment (kazyshr) [inline ecl/sss/v1/approval_requests/testing/fixtures.go:14]: Why not test string body as well as json?

File: ecl/sss/v1/approval_requests/testing/fixtures.go (hunk 4 above)
Comment (ghost) [reply ecl/sss/v1/approval_requests/testing/fixtures.go]: My understanding is that API resp is byte before extracting, but not 100% sure..

Comment (shimisho) [reply ecl/sss/v1/approval_requests/testing/fixtures.go]: Actions.Body contains json.RawMessage, so do the same for actionsBody.

File: ecl/sss/v1/approval_requests/testing/fixtures.go (hunk 5 above)
Comment (shimisho) [reply ecl/sss/v1/approval_requests/testing/fixtures.go:14]: I wrote test code with actions.body as a string.

File: ecl/sss/v1/approval_requests/testing/fixtures.go [hunk 6]
```diff
@@ -1,99 +1,107 @@
...
 	idApprovalRequest2,
+	stringBody,
 )
 
 var expectedApprovalRequestsSlice = []ar.ApprovalRequest{
 	firstApprovalRequest,
 	secondApprovalRequest,
 }
 
-var actionForApproveRequest1 = ar.Action{
```
Comment (shimisho) [inline ecl/sss/v1/approval_requests/testing/fixtures.go:89]: Removed because it is not used.

**Comment by shimisho** [comment]
Please check again.

File: ecl/sss/v1/approval_requests/testing/requests_test.go [hunk 7]
```diff
@@ -25,11 +25,7 @@ func TestListApprovalRequest(t *testing.T) {
 	})
 
 	count := 0
-	opts := ar.ListOpts{
```
Comment (shimisho) [inline ecl/sss/v1/approval_requests/testing/requests_test.go:28]: Since it is not necessary to narrow down the data by the option, I deleted it.

**Review by kazyshr — APPROVED** [review]
Great!

**Pull request merged**